src/data-access/mutations/useAddReactionsComment.ts

```diff
@@ -98,3 +80,60 @@ export const useEditComment = () =>
       console.log('Error while editing comment: ', err.message)
     },
   })
+
+const addReaction = async (reaction: AddReaction): Promise<SubmitSuccess> => {
+  const { data } = await axios.put(API.PUT.addReactionToPost(reaction), reaction)
+  return data
+}
+export const useAddReaction = () => {
+  const { user } = useUserContext()
+  return useMutation<SubmitSuccess, AxiosError<{ errors: string[] }>, AddReaction>(addReaction, {
+    onMutate: async ({ reaction, postId }) => {
+      await queryClient.cancelQueries([QueryKeys.POSTS])
+      const allPosts: FeedPost[] = queryClient.getQueryData([QueryKeys.POSTS]) || []
+      const postIndex = allPosts?.findIndex((post) => post.id === postId)
+      const reactionToUpdate = allPosts[postIndex].reactions.filter(
+        (oldReaction) => oldReaction.type === reaction.type
+      )
+      const typesOfReactionsInPost = allPosts[postIndex].reactions.map((reaction) => reaction.type)
+      const allPostReactions = allPosts[postIndex].reactions
+
+      if (!reactionToUpdate[0]?.users.includes(user?.id || '')) {
+        let updatedReactions: Reaction[] = allPostReactions.map((oldReaction) => {
+          if (oldReaction.type === reaction.type)
+            return { ...oldReaction, users: [...oldReaction.users, user?.id || ''] }
+          return oldReaction
+        })
+        if (!typesOfReactionsInPost.includes(reaction.type))
+          updatedReactions = [...updatedReactions, reaction]
+        const updatedPost: FeedPost = {
+          ...allPosts[postIndex],
+          reactions: updatedReactions,
+        }
+        allPosts[postIndex] = updatedPost
+        queryClient.setQueryData([QueryKeys.POSTS], allPosts)
+      }
+    },
+    onSuccess: (payload) => {
+      queryClient.setQueryData<FeedPost[]>([QueryKeys.POSTS], (data) => {
+        if (!data) throw new Error('No posts found!')
+        const allPosts = data
+        const postIndex = allPosts?.findIndex((post) => post.id === payload.post.id)
+        if (postIndex !== -1 && allPosts) {
+          const filtered = payload.post.reactions.filter((reaction) => reaction.users.length > 0)
+          payload.post.reactions = filtered
+          allPosts[postIndex] = payload.post
+        }
+
+        if (allPosts?.length) return [...allPosts]
+        return [payload.post]
+      })
+    },
```

Comment:
```suggestion
```
You could place invalidateQueries in onSuccess, but onSettled will handle it anyway